internal/jennies/golang/tmpl.go

```diff
@@ -63,6 +63,7 @@ func initTemplates(extraTemplatesDirectories []string) *template.Template {
 		}),
 		template.Funcs(map[string]any{
 			"formatPackageName": formatPackageName,
+			"formatObjectName":  formatObjectName,
```

Comment:
Mainly for my own understanding: why do we add `formatObjectName` but not `formatFieldName` as template function?
Or maybe I'm failing to "parse" the templates we emit, to find out such difference in tmpl function usage 😇 .

Reply:
I started using these `formatX` functions to be consistent with how it's done in other languages than Go: cf https://github.com/grafana/cog/blob/ecbfa726391eb1771f0d08b9bcc057fd6dff07c1/internal/jennies/php/tmpl.go#L25-L34

But since this PR was getting a bit big already, I thought I would change only what I immediately needed and keep working on this in another PR :)

Reply:
The follow-up to this is here: https://github.com/grafana/cog/pull/635 :)